openfisca_us/api/microsimulation.py

```diff
@@ -30,65 +30,100 @@ def apply_reforms(self, reforms: list) -> None:
 
     def load_dataset(self, dataset, year):
         self.system = openfisca_us.CountryTaxBenefitSystem()
-        self.apply_reforms(
-            (BaseCPS.input_reform_from_year(year), self.reforms)
-        )
+        self.apply_reforms(self.reforms)
         builder = SimulationBuilder()
         builder.create_entities(self.system)
 
-        base_cps = BaseCPS.load(year)
+        data = dataset.load(year)
 
-        builder.declare_person_entity("person", base_cps[f"person_id/{year}"])
+        builder.declare_person_entity("person", data[f"person_id"])
 
         for group_entity in ("tax_unit", "family", "spm_unit", "household"):
-            primary_keys = np.array(base_cps[f"{group_entity}_id/{year}"])
+            primary_keys = np.array(data[f"{group_entity}_id"])
             group = builder.declare_entity(group_entity, primary_keys)
-            foreign_keys = np.array(
-                base_cps[f"person_{group_entity}_id/{year}"]
-            )
+            foreign_keys = np.array(data[f"person_{group_entity}_id"])
             builder.join_with_persons(
                 group, foreign_keys, np.array(["member"] * len(foreign_keys))
             )
 
         model = builder.build(self.system)
 
-        for variable in base_cps.keys():
+        for variable in data.keys():
             if variable in self.system.variables:
-                for input_period in base_cps[variable].keys():
-                    model.set_input(
-                        variable,
-                        input_period,
-                        np.array(base_cps[variable][input_period]),
-                    )
+                model.set_input(
+                    variable,
+                    year,
+                    np.array(data[variable]),
+                )
 
         self.simulation = model
 
-    def calc(self, variable, period=2020, weighted=True):
+    def map_to(
+        self, arr: np.array, entity: str, target_entity: str, how: str = None
+    ):
+        entity_pop = self.simulation.populations[entity]
+        target_pop = self.simulation.populations[target_entity]
+        GROUP_ENTITIES = ("tax_unit", "family", "spm_unit", "household")
+        if entity == "person" and target_entity in GROUP_ENTITIES:
+            if how and how not in (
+                "sum",
+                "any",
+                "min",
+                "max",
+                "all",
+                "value_from_first_person",
```

Comment:
not just `first`?

Reply:
Unfortunately not, these are [openfisca-implemented functions](https://github1s.com/openfisca/openfisca-core/blob/HEAD/openfisca_core/populations/group_population.py) rather than pandas function.